alternator/conditions.cc

```diff
@@ -125,6 +125,13 @@ static bool check_BEGINS_WITH(const rjson::value* v1, const rjson::value& v2) {
     return val1.substr(0, val2.size()) == val2;
 }
 
+static void verify_operands(const rjson::value& array, rapidjson::SizeType expected, const rjson::value& name) {
+    if (array.Size() != expected) {
+        throw api_error("ValidationException",
+                        format("{} operator requires one element in AttributeValueList", name));
```

Comment:
This has the word "one" in the message instead of the expected value...
Which leads me to think - if every operator has a different requirement (some require one, some two, some many), I think it would be better and clearer to simply keep the two line if() as we had before this patch? You're not really avoiding any code duplication, since every one of these cases has a different comparison and a different message anyway...

Reply:
> This has the word "one" in the message instead of the expected value...

Sorry, fixed now.

> Which leads me to think - if every operator has a different requirement (some require one, some two, some many), I think it would be better and clearer to simply keep the two line if()

It's two lines only if you overlook the third.  Why repeat nearly-identical three lines everywhere when it's trivial to fix this message, as the latest commit just did.

Whether different operators require different `verify_operator` arguments is a red herring; the point is that `verify_operator` factors out common code and yes, it avoids code duplication.